### PR TITLE
[4.4] Phan: Fixing banners component

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This fixes reported issues from phan (static code analysis) in the backend banners component.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_banners/src/View/Banner/HtmlView.php [hunk 1]
```diff
@@ -70,11 +69,9 @@ class HtmlView extends BaseHtmlView
      */
     public function display($tpl = null): void
     {
-        /** @var BannerModel $model */
-        $model       = $this->getModel();
-        $this->form  = $model->getForm();
-        $this->item  = $model->getItem();
-        $this->state = $model->getState();
+        $this->form  = $this->get('Form');
```
Comment (joomdonation) [inline administrator/components/com_banners/src/View/Banner/HtmlView.php]: Personal, I prefer the original code: It is concrete,  faster, and more important, IDE friendly. When we use `$model->getForm()`, IDE could help us find the method implementation easily, unlike when we use `$this->get('Form');`

File: administrator/components/com_banners/src/Extension/BannersComponent.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function boot(ContainerInterface $container)
     /**
      * Returns the table for the count items functions for the given section.
      *
-     * @param   string  $section  The section
+     * @param   string?  $section  The section
```
Comment (wilsonge) [inline administrator/components/com_banners/src/Extension/BannersComponent.php]:
```suggestion
     * @param   ?string  $section  The section
```

File: administrator/components/com_banners/src/View/Banner/HtmlView.php (hunk 1 above)
Comment (wilsonge) [reply administrator/components/com_banners/src/View/Banner/HtmlView.php]: Yes this version with types is something we agreed to role out further through core. not sure if it got completed or not but this was a deliberate decision and something @roland-d was working on at one point.

Comment (Hackwar) [reply administrator/components/com_banners/src/View/Banner/HtmlView.php]: ok, will revert those changes.

File: administrator/components/com_banners/src/Helper/BannersHelper.php [hunk 3]
```diff
@@ -90,6 +90,7 @@ public static function updateReset()
 
             switch ($purchaseType) {
                 case 1:
+                default:
```
Comment (wilsonge) [inline administrator/components/com_banners/src/Helper/BannersHelper.php]: I'm not saying this is wrong - but feels like there's other logic issues getting us here? The default value for this is 0 which implies that we should be handling that case in the component fallbacks above?

Comment (Hackwar) [reply administrator/components/com_banners/src/Helper/BannersHelper.php]: I reverted this for now.

**Pull request merged**